crates/driver/src/print_build.rs

```diff
@@ -259,48 +259,51 @@ where
     for graph_node in graph_nodes {
         let node_file = opt.working_directory.join(&graph_node.node_label);
         let node_file_name = to_file_name(&node_file);
+        let mut extra_kv_pairs: HashMap<String, Vec<String>> = HashMap::default();
+        let (build_config, use_rglob) = if path_type == "test" {
+            (&module_config.build_config.test, true)
+        } else {
+            (&module_config.build_config.main, false)
+        };
+
+        let build_config = if let Some(bc) = build_config {
+            bc
+        } else {
+            return Err(anyhow!(
+                "unable to find build configuration for {:?}",
+                graph_node
+            ));
+        };
+
+        let include_file_extension = build_config.include_file_extension;
         let target_name = if !opt.no_aggregate_source {
             base_name.clone()
         } else {
-            node_file_name.replace(".", "_")
-        };
-        let mut extra_kv_pairs: HashMap<String, Vec<String>> = HashMap::default();
+            to_name_from_file_name(&node_file_name, include_file_extension)?
+        }; 
 
         fn add_non_empty(
             opt: &'static Opt,
             key: &str,
             labels: &Vec<String>,
             extra_kv_pairs: &mut HashMap<String, Vec<String>>,
+            include_file_extension: bool,
```

Comment:
could we avoid boolean blindness with a small Enum to represent the semantics:
```rust
enum TargetNameStrategy { Directory, DirectoryFileType }
```
or something